appointment/upload-documents/uploadDocuments.php

```diff
@@ -73,21 +73,142 @@ class="submit dcf-btn dcf-btn-primary"
+						<div ng-show="consentForm.$submitted">
+							<label class="error" ng-show="consentForm.consentToReview.$error.required">This field is required.</label>
+						</div>
+					</li>
 
-				<div class="dcf-input-checkbox">
-					<input id="agree-to-virtual-preparation-checkbox" type="checkbox" ng-model="agreeToVirtualPreparationCheckbox.checked" value="false">
-					<label for="agree-to-virtual-preparation-checkbox">I agree to have my tax return prepared virtually</label>
-				</div>
+					<p class="dcf-bold">Virtual Consent Disclosure:</p>
+					<p>If you agree to have your tax return prepared and your tax documents handled in the above manner,
+						your signature and/or agreement is required on this document. Signing this document means that
+						you are agreeing to the procedures stated above for preparing a tax return for you. (If this
+						is a Married Filing Joint return both spouses must sign and date this document.) If you chose
+						not to sign this form, we may not be able to prepare your tax return using this process. Since
+						we are preparing your tax return virtually, we have to secure your consent agreeing to this 
+						process. If you consent to use these non-IRS virtual systems to disclose or use your tax return
+						information, Federal law may not protect your tax return information from further use or distribution
+						in the event these systems are hacked or breached without our knowledge. If you agree to the disclosure
+						of your tax return information, your consent is valid for the amount of time that you specify. If
+						you do not specify the duration of your consent, your consent is valid for one year from the date
+						of signature. If you believe your tax return information has been disclosed or used improperly in a
+						manner unauthorized by law or without your permission, you may contact the Treasury Inspector General
+						for Tax Administration (TIGTA) by telephone at 1-800-366-4484, or by e-mail at complaints@tigta.treas.gov .
+						While the IRS is responsible for providing oversight requirements to Volunteer Income Tax Assistance (VITA) 
+						and Tax Counseling for the Elderly (TCE) programs, these sites are operated by IRS sponsored partners who
+						manage IRS site operations requirements and volunteer ethical standards. In addition, the locations of these 
+						ites may not be in or on federal Property. </p>
+					<li class="form-radio" id="consentVirtual">
+						<label class="form-required">I am agreeing to use this site's Virtual VITA/TCE Process</label>
+						<div>
+							<div class="dcf-btn-group" data-toggle="buttons">
+								<label class="dcf-btn dcf-btn-secondary" 
+									name="consentToVirtual" 
+									ng-model="consentData.virtualConsent" 
+									uib-btn-radio="1" 
+									required>Yes
+								</label>
+								<label class="dcf-btn dcf-btn-secondary" 
+									name="consentToVirtual" 
+									ng-model="consentData.virtualConsent" 
```

Comment:
What happens if someone says no to this field? Can we display a red error saying "We cannot prepare your tax return without your affirmative consent" or whatever? It seems strange to me that saying "No" here just disables the submit button and doesn't give user any reason why

Reply:
Yeah, good idea